log.js

```diff
@@ -13,18 +13,234 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  *
- * @fileoverview Controls for the logging frame.
+ * @fileoverview Handles the button control events and appends logs for the actual log window.
  */
-document.addEventListener('DOMContentLoaded', function() {
+document.addEventListener('DOMContentLoaded', function () {
   const clearButton = document.querySelector('#clear-button');
-  clearButton.addEventListener('click', () => {
-    EmeLogWindow.instance.clear();
-  });
+  if (clearButton != null) {
+    clearButton.addEventListener('click', () => {
+      chrome.runtime.sendMessage({ type: 'EME_LOGGER_CLEAR' });
+
+      const list = document.getElementById('eme-log');
+      while (list.hasChildNodes()) {
+        list.removeChild(list.firstChild);
+      }
+    });
+  }
 
   const downloadButton = document.querySelector('#download-button');
-  downloadButton.addEventListener('click', () => {
-    // Set the download URI to a freshly-generated one that contains the
-    // current text from the log window.
-    downloadButton.href = EmeLogWindow.instance.getTextLogUri();
-  });
+  if (downloadButton != null) {
+    downloadButton.addEventListener('click', async function () {
+      // Get the text logs from the log window.
+      const response = await chrome.runtime.sendMessage({ type: 'EME_LOGGER_GET_TEXT_LOGS' });
+      const blob = new Blob([response.textLogs], { type: 'text/plain' });
+
+      // Trigger a download
+      chrome.downloads.download({
+        url: URL.createObjectURL(blob),
+        filename: 'EMELogFile.txt'
+      });
+    });
+  }
 });
+
+// NOTE: These APIs are not defined in our test environment, but should always
+// be present when this is run as a Chrome extension.
+if (chrome.runtime !== undefined) {
+  /**
+   * Listens for messages from the content script to append a log item to the
+   * current frame and log file.
+   */
+  chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+    const textLog = appendLog(request.log);
+    chrome.runtime.sendMessage({ type: 'EME_LOGGER_APPEND_LOG', data: textLog });
+  });
+}
+
+/**
+ * @param {Object} The serialized log to format in HTML.
+ * @return {string} The text log.
+ */
+function appendLog(log) {
```

Comment:
I see a few issues with this function.

1. This being global confused me when I saw "appendLog()" in the tests.  I couldn't tell where it came from.  (When John moves it to a class or namespace later, that will have to be more explicit.)
2. The name of this method made sense when it was part of EmeLogWindow (when it both formatted and appended the logs), but now I'm not so sure.  It _does_ still append logs to the DOM, but it also returns a string for the text-only version.  In the tests, `EmeLogWindow.instance.appendLog(appendLog(...` is still confusing to read, because the name appendLog() doesn't explain what it's doing.

I'm feeling picky about this thing, but it's not really part of the Manifest V3 cleanup.  How about you leave a TODO(joeyparrish): cleanup here, and I'll restructure/rename this part after this is merged.

Sound good?

Reply:
This part was the most concerned area while moving the logic around and updating the test. So your concern does totally make sense to me. I've added `// TODO(joeyparrish): cleanup here, and I'll restructure/rename this part after this is merged.` before the `function appendLog() {` line. Thank you for your thought.